### ISSUE TITLE
Liana on pseduo bulk-sc data data

### ISSUE THREAD

**Issue by Marwansha** [saezlab/liana-py#96]
Hey,
I want to know if it's best practice to always use single cell data on liana to compute the l-r ccc results, as I saw in the tutorial of differential expression, after computing differential expression on pseduobulk data, Liana was run on the adata object, and I want to ask if it's best practise and if it's ok to run it on the pseduobulk data?

Thanks
Marwan

**Comment by dbdimitrov** [comment]
Hi @Marwansha,

From my knowledge, it is the current best practice to perform differential testing between single-cell samples at the pseudobulk level. 

A couple of reference on the topic:

https://www.nature.com/articles/s41576-023-00586-w
https://www.nature.com/articles/s41467-021-21038-1

**Comment by Marwansha** [comment]
Hey, I think maybe you misunderstood my question, I meant running Liana on the pseduobulk computed anndata object, Not the differential expression

**Comment by dbdimitrov** [comment]
Hi @Marwansha,

Sorry but not sure I exactly follow. Can you elaborate which analysis and in what sense on pseudobulks?

One can also say that average expression per cluster is a "pseudobulk" (which is what the vast majority of CCC methods approach).

**Comment by Marwansha** [comment]
sorry if i wasn't clear again, 

my question is about generation the ligand-receptor interactions df
for example simply here for a single cell data object :
`li.mt.rank_aggregate(adata, groupby='celltype', expr_prop=0.1, verbose=True)`

my question is if i run the run aggreagate on the pseudobulk anndata object rather than the single cell object?
`li.mt.rank_aggregate(**_pdata_**, groupby='celltype', expr_prop=0.1, verbose=True)`
the one generated by decoupler, which here liana will treat each individal_celltype_ as  1 observation , so if we have 10 ind, for 1 celltype we got 10 observations, while in single ell data we got no of cells are the observation for a celltype
```
pdata = dc.get_pseudobulk(
    adata,
    groups_col="celltype",
    layer='counts',
    mode='sum',
    min_cells=10,
    min_counts=10000
)
pdata
```

i thought running liana on the pseudobulk aggregated anndata object  make sense for mycase and get results that are less nosiy but i was not sure if this was tested before or not

Thanks

**Comment by dbdimitrov** [comment]
Hi @Marwansha,

Sorry for the delay, I was on away.

Hmm. This is a really interesting approach, though not standard.
If you have normalized (total + log1p) the summed counts, I see nothing wrong it with.

It only changes a bit the interpretation, since instead of comparing means across cells, you are comparing means across sample pseudobulks. 

Just to share my intuition with this, think of CellPhoneDB. You get a mean between the averaged ligand and receptor expression per cluster (`lr_mean`), and you get a p-value where the averaging is done on permuted cell labels (`cpdb_pvals`). In your case, I believe the `lr_mean` ranking shouldn't change too much whether you do it on pseudobulks or at the single-cell level. However, the p-values should be quite different (since you are shuffling pseudobulks per sample) and would likely be a bit more conservative.

In short, it can make sense depending on your data :)

**Comment by dbdimitrov** [comment]
PS. A major motivation of mine when writing liana-py was to make it flexible, so I'm glad to see when it's used in ways beyond the tutorials.

**Comment by Marwansha** [comment]
Thank you very much for your response. In fact, I am trying to benchmark and compare the different results that come from computing the CCC (cell-cell communication) on single-cell or pseudobulk objects. From my perspective, and from a ground truth point of view (considering some ligand-receptor interactions that exist in one group and not in the other, which I know from literature and previous work), it seems that using the pseudobulk data makes it cleaner and easier to discern.

Would you be interested in having a short meeting? Maybe I can show you my data (I can share it too), so I can get some insights from your point of view on which approach makes more sense.

Thanks
Marwan
